### DISCUSSION TITLE
New feature of adding Watermarks

### DISCUSSION THREAD

**Discussion by justemu** [outline/outline#2259] [Ideas]
Some of the team's knowledge collections are meant to be kept secret with the organization. Therefore the idea of an option to adding  watermarks containing the datetime and the name and profile of the viewer, would be a great one for prevetion of unmeant information leakage. This kind of information should also be kept from downloading and view only.

How about that?

**Comment by tommoor** [comment]
Hi @justemu – downloads are just plain text files at the moment, I'm not aware of a way to embed watermarks in such files that couldn't be trivially removed. Do you have any examples of software that does this?

**Reply by justemu** [reply]
There is an online knowledge base software by alibaba named "[yuque](https://www.yuque.com/about/products?language=en-us)". PLS follow the link to check out.
 
It's not open-source, but with similar experience as outline. I don't know how it manage to add wartermarks globally upon user settings. Maybe... by adding a layer of picture with opacity in the rendering process?  on the server side?

I just realised that the two similar-looking products might be of completely different artitectures... Since online is super-fast featured and doing most of the work on the client side, the plain text data should have been sent to the client-side straightforwardly in the very beginning.
